packages/vats/src/core/basic-behaviors.js

```diff
@@ -224,10 +224,41 @@ export const makeAddressNameHubs = async ({
 };
 harden(makeAddressNameHubs);
 
+const WALLET_STORAGE_PATH = 'wallet';
+
 /** @param {BootstrapSpace} powers */
-export const makeClientBanks = async ({ consume: { client, bankManager } }) => {
+export const makeClientBanks = async ({
+  consume: {
+    agoricNames,
+    board,
+    namesByAddress,
+    namesByAddressAdmin,
+    client,
+    chainStorage,
+    bankManager,
+    zoe,
+  },
+  installation: {
+    consume: { smartWallet },
+  },
+}) => {
+  const chainStoragePresence = await chainStorage;
+  const walletChainStorage = await (chainStoragePresence &&
+    E(chainStoragePresence).getChildNode(WALLET_STORAGE_PATH));
+  const marshaller = E(board).getPublishingMarshaller();
   return E(client).assignBundle([
-    address => ({ bank: E(bankManager).getBankForAddress(address) }),
+    address => {
+      const bank = E(bankManager).getBankForAddress(address);
+      const myAddressNameAdmin = E(namesByAddressAdmin).lookupAdmin(address);
+      const myWallet = E(zoe).startInstance(
+        smartWallet,
+        {},
+        { agoricNames, bank, namesByAddress, myAddressNameAdmin, board },
```

Comment:
Both the `smartWallet` contract an `ag-solo` use `myAddressNameAdmin` to set up `depositFacet`. There can be only one. This is why we see 0 invitations in xnet https://github.com/Agoric/agoric-sdk/issues/5513#issuecomment-1156612543 .